src/InvalidTypeException.java

```diff
@@ -0,0 +1,16 @@
+@SuppressWarnings("serial")
+public class InvalidTypeException extends WordsException {
+	
+	private String expectedType;
+	private String receivedType;
+	
+	public InvalidTypeException(int lineNo, String expected, String received) {
+		super(lineNo);
+		this.expectedType = expected;
+		this.receivedType = received;
+	}
+
+	public String toString() {
```

Comment:
I'll add in the override annotation to make it explicit (though Java overrides anyway without the annotation).